File: benchmarks/poly/loopsum/loopsum.pl
use Time::HiRes qw(clock_gettime CLOCK_MONOTONIC);

sub main {
  my $_t0 = clock_gettime(CLOCK_MONOTONIC);
  my $acc = 0;
  for my $i (1 .. 10000000) {
    $acc = ($acc + $i * $i) % 1000000007;
  }
  print $acc . "\n";
  printf STDERR "bench_ns=%.0f\n", (clock_gettime(CLOCK_MONOTONIC) - $_t0) * 1e9;
}

main();

File: benchmarks/poly/loopsum/loopsum.rb
def main
  _t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  acc = 0
  (1..10000000).each do |i|
    acc = (acc + i * i) % 1000000007
  end
  puts acc
  $stderr.puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - _t0}"
end

main
